packages/daimo-common/src/daimoLink.ts

```diff
@@ -138,32 +158,49 @@ function parseDaimoLinkInner(link: string): DaimoLink | null {
       return { type: "request", requestId, recipient, dollars };
     }
     case "note": {
-      let previewSender: string, previewDollars: `${number}`, ephem: string;
-      if (parts.length === 2) {
-        // backcompat with old notes
-        [previewSender, previewDollars] = ["unknown", "0.00"];
-        ephem = parts[1];
-      } else if (parts.length === 4) {
-        previewSender = parts[1];
+      if (parts.length === 4) {
+        // backcompat with old links
+        const previewSender = parts[1];
         const parsedDollars = zDollarStr.safeParse(parts[2]);
         if (!parsedDollars.success) return null;
-        previewDollars = parseFloat(parsedDollars.data).toFixed(2) as DollarStr;
-        ephem = parts[3];
+        const previewDollars = parseFloat(parsedDollars.data).toFixed(
+          2
+        ) as DollarStr;
+        const hashParts = parts[3].split("#");
+        if (hashParts.length > 2) return null;
+        const ephemeralOwner = getAddress(hashParts[0]);
+        const ephemeralPrivateKey =
+          hashParts.length < 2 ? undefined : zHex.parse(hashParts[1]);
+        return {
+          type: "note",
+          previewSender,
+          previewDollars,
+          ephemeralOwner,
+          ephemeralPrivateKey,
+        };
       } else return null;
+    }
+    case "n": {
```

Comment:
maybe `claim` for readability

comparable: `https://peanut.to/claim#?c=8453&v=v4&i=38&p=<key>`

Reply:
kinda like the shorter link though, we can also make `account` -> `u` and `request` -> `r` in future like reddit, which one do you prefer?

Reply:
sounds good, go for it